packages/core/monaco-editor/src/components/MonacoEditor.vue

```diff
@@ -1,16 +1,300 @@
+
+/**
+ * Computed property to determine if the editor is empty.
+ * @returns {boolean}
+ */
+const isEditorEmpty = computed<boolean>(() => monacoEditor.editorStates.editorStatus === 'ready' && !monacoEditor.editorStates.hasContent)
+
+const monacoEditor = useMonacoEditor(editor, {
+  language,
+  code: model,
+  theme: editorTheme.value,
+  monacoOptions: options,
+  onChanged: (content: string): void => {
+    model.value = content
+  },
+})
 </script>
 
 <style lang="scss" scoped>
-.kong-ui-public-monaco-editor {
-  // Add component styles as needed
+.monaco-editor-container {
+  background: $kui-color-background;
+  display: flex;
+  flex-direction: column;
+  height: 100%;
+  overflow: hidden;
+  position: relative;
+  width: 100%;
+
+  &.loading {
+    pointer-events: none;
+    user-select: none;
+
+    .monaco-editor-target {
+      filter: blur(2px);
+      opacity: 0;
+      pointer-events: none;
+      user-select: none;
+    }
+
+  }
+
+}
+
+.monaco-editor-target {
+  height: 100%;
+  overflow: hidden;
+  position: relative;
+  transition: all $kui-animation-duration-20 ease-in-out;
+  width: 100%;
+
+  // Customize the editor's light theme
+  &.light {
+    :deep(.monaco-editor) {
+      // Customize monaco editor styles via `--vscode-` variables
+      /* stylelint-disable */
+      // Editor
+      --vscode-editor-background: #{$kui-color-background};
+      --vscode-editorGutter-background: #{$kui-color-background};
+      --vscode-editorLineNumber-activeForeground: #{$kui-color-text-primary};
+      // Suggestions
+      --vscode-editorSuggestWidget-background: #{$kui-color-background};
+      --vscode-editorSuggestWidget-border: #{$kui-color-border};
+      // Context menu
+      --vscode-menu-background: #{$kui-color-background};
+      --vscode-menu-border: #{$kui-color-border};
+      --vscode-menu-separatorBackground: #{$kui-color-border};
+      // Other
+      --vscode-focusBorder: #{$kui-color-text-neutral};
+      --vscode-input-background: #{$kui-color-background};
+      --vscode-sash-hoverBorder: #{$kui-color-border-primary};
+      /* stylelint-enable */
+
+      // Modify the editor's search box styles
+      .find-widget {
+        background: $kui-color-background;
+        border-bottom: $kui-border-width-10 solid $kui-color-border-neutral-weaker;
+
+        // the pane to resize the search box
+        .monaco-sash {
+          background-color: $kui-color-background-neutral-weaker;
+        }
+
+        // Modify the search input
+        .monaco-inputbox {
+          background-color: $kui-color-background !important;
+          border: $kui-border-width-10 solid $kui-color-border-neutral-weaker !important;
+        }
+      }
+    }
+  }
+
+  // TODO: add dark theme once the tokens are ready
+  // Customize the editor's dark theme
+  // &.dark {
+  // }
+
+  /** !Important: Only put non-color related overrides in this block */
+  :deep(.monaco-editor) {
```

Comment:
nit: please add comments in the style rules here to explain what we're customizing